### PR TITLE
optimized a couple loss functions

### PR DESCRIPTION
**Pull request by sidd-27**
One-hot encoding was removed by using gather, which directly selects the true-class probability without expanding to all classes. also in the other functions used more optimal variable handling and pytorch functions, optimization of 4x in some cases

https://colab.research.google.com/drive/1DusGb0krES6hIHKfXAJwv_zzyJlSgfkb?usp=sharing

heres the code which benchmarks and tests these changes

### CODE REVIEW

**Comment by cjpurackal** [comment]
can you please add the names of the loss functions in the PR ?
